src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.test.tsx

```diff
@@ -380,6 +397,69 @@ describe('With Registered Authentication', () => {
       }
     );
 
-    expect(mockCredentialForm).toHaveBeenCalled();
+    expect(mockedCredentialForm).toHaveBeenCalled();
+  });
+
+  test('should update the form with registered auth type on click save changes', async () => {
+    const mockedCredentialForm = jest.fn();
+    const mockedSubmitHandler = jest.fn();
+    const authMethodToBeTest = {
+      name: 'Some Auth Type',
+      credentialSourceOption: {
+        value: 'Some Auth Type',
+        inputDisplay: 'some input',
+      },
+      credentialForm: mockedCredentialForm,
+      crendentialFormField: {},
+    } as AuthenticationMethod;
+
+    const mockedContext = mockManagementPlugin.createDataSourceManagementContext();
+    mockedContext.authenticationMethodRegistery = new AuthenticationMethodRegistery();
+    mockedContext.authenticationMethodRegistery.registerAuthenticationMethod(authMethodToBeTest);
+
+    component = mount(
+      wrapWithIntl(
+        <EditDataSourceForm
+          existingDataSource={mockDataSourceAttributesWithRegisteredAuth}
+          existingDatasourceNamesList={existingDatasourceNamesList}
+          onDeleteDataSource={jest.fn()}
+          handleSubmit={mockedSubmitHandler}
+          handleTestConnection={jest.fn()}
+          displayToastMessage={jest.fn()}
+        />
+      ),
+      {
+        wrappingComponent: OpenSearchDashboardsContextProvider,
+        wrappingComponentProps: {
+          services: mockedContext,
+        },
+      }
+    );
+
+    await new Promise((resolve) =>
+      setTimeout(() => {
+        updateInputFieldAndBlur(component, descriptionFieldIdentifier, '');
+        expect(
+          // @ts-ignore
+          component.find(descriptionFormRowIdentifier).first().props().isInvalid
+        ).toBeUndefined();
+        resolve();
+      }, 100)
+    );
+    await new Promise((resolve) =>
+      setTimeout(() => {
+        /* Updated description*/
+        updateInputFieldAndBlur(component, descriptionFieldIdentifier, 'testDescription');
+        expect(
+          // @ts-ignore
+          component.find(descriptionFormRowIdentifier).first().props().isInvalid
+        ).toBeUndefined();
```

Comment:
same as above. On line 452, you set `testDescription` to description. But how expected behaviour is same. It is expected to be same (if it doesn't care whether is empty or not), why need to test?

Reply:
Thanks for the comments~

The reason why I have those two checks is trying to mock following customer action: "Editing description of the datasource and then click on "Submit Change" button".

But since that testing scenario already been handled before and introducing confusions here, let me remove it.